src/bio/bio_device.c

```diff
@@ -773,30 +776,44 @@ led_device_action(void *ctx, struct spdk_pci_device *pci_device)
 		return;
 	}
 
-	/* Set the LED to the new state */
-	rc = spdk_vmd_set_led_state(pci_device, led_state_daos2spdk(opts->led_state));
-	if (spdk_unlikely(rc != 0)) {
-		D_ERROR("Failed to set the VMD LED state on %s (%s)\n", addr_buf,
-			spdk_strerror(-rc));
-		opts->status = -DER_NOSYS;
-		return;
-	}
+	if (vmd_on) {
+		/* Set the LED to the new state */
+		rc = spdk_vmd_set_led_state(pci_device, led_state_daos2spdk(opts->led_state));
+		if (spdk_unlikely(rc != 0)) {
+			D_ERROR("Failed to set the VMD LED state on %s (%s)\n", addr_buf,
+				spdk_strerror(-rc));
+			opts->status = -DER_NOSYS;
+			return;
+		}
 
-	rc = spdk_vmd_get_led_state(pci_device, &cur_led_state);
-	if (rc != 0) {
-		D_ERROR("Failed to get the VMD LED state on %s (%s)\n", addr_buf,
-			spdk_strerror(-rc));
-		opts->status = -DER_NOSYS;
-		return;
+		rc = spdk_vmd_get_led_state(pci_device, &cur_led_state);
+		if (rc != 0) {
+			D_ERROR("Failed to get the VMD LED state on %s (%s)\n", addr_buf,
+				spdk_strerror(-rc));
+			opts->status = -DER_NOSYS;
+			return;
+		}
+		d_led_state = led_state_spdk2daos(cur_led_state);
+	} else {
+		/* Set current state to expected if no VMD */
+		d_led_state = opts->led_state;
 	}
-	d_led_state = led_state_spdk2daos(cur_led_state);
 
 	/* Verify the correct state is set */
 	if (d_led_state != opts->led_state) {
 		D_ERROR("Unexpected LED state on %s, want %s got %s\n", addr_buf,
 			LED_STATE_NAME(opts->led_state), LED_STATE_NAME(d_led_state));
 		opts->status = -DER_INVAL;
+		return;
```

Comment:
I know it's old code, but I'm not sure `-DER_INVAL` is the right rc here. Usually `-DER_INVAL` means the caller entered something invalid, whereas this is a case where the operation failed at some level below us.